jcl/src/java.base/share/classes/java/lang/System.java

```diff
@@ -176,18 +251,82 @@ static void afterClinitInitialization() {
+		Charset stderrCharset = getCharset(props.getProperty("sun.stderr.encoding"), null); //$NON-NLS-1$
+		/*[ELSE]*/
+		Charset consoleCharset = Charset.defaultCharset();
+		String stdoutCharset = getCharsetName(props.getProperty("sun.stdout.encoding"), consoleCharset); //$NON-NLS-1$
+		String stderrCharset = getCharsetName(props.getProperty("sun.stderr.encoding"), consoleCharset); //$NON-NLS-1$
+		/*[ENDIF] JAVA_SPEC_VERSION >= 11 */
+		/*[ENDIF] Sidecar18-SE-OpenJ9 | JAVA_SPEC_VERSION >= 11 */
+
+		/*[IF (Sidecar18-SE-OpenJ9 | JAVA_SPEC_VERSION >= 11) & !(PLATFORM-mz31 | PLATFORM-mz64)]*/
+		if (stderrCharset != null) {
+			/*[IF JAVA_SPEC_VERSION >= 11]*/
+			hasSetErrEncoding = true;
+			setErr(new PrintStream(new BufferedOutputStream(new FileOutputStream(FileDescriptor.err)), true, stderrCharset));
+			/*[ELSE]*/
+			try {
+				setErr(new PrintStream(new BufferedOutputStream(new FileOutputStream(FileDescriptor.err)), true, stderrCharset));
+			} catch (UnsupportedEncodingException e) {
+				// not possible as the Charset name was validated first
+				throw new InternalError("For encoding " + stderrCharset, e); //$NON-NLS-1$
+			}
+			/*[ENDIF] JAVA_SPEC_VERSION >= 11 */
+		} else {
+			setErr(new PrintStream(new BufferedOutputStream(new FileOutputStream(FileDescriptor.err)), true));
+		}
+		if (stdoutCharset != null) {
+			/*[IF JAVA_SPEC_VERSION >= 11]*/
+			hasSetOutEncoding = true;
+			setOut(new PrintStream(new BufferedOutputStream(new FileOutputStream(FileDescriptor.out)), true, stdoutCharset));
+			/*[ELSE]*/
+			try {
+				setOut(new PrintStream(new BufferedOutputStream(new FileOutputStream(FileDescriptor.out)), true, stdoutCharset));
+			} catch (UnsupportedEncodingException e) {
+				// not possible as the Charset name was validated first
+				throw new InternalError("For encoding " + stdoutCharset, e); //$NON-NLS-1$
+			}
+			/*[ENDIF] JAVA_SPEC_VERSION >= 11 */
+		} else {
+			setOut(new PrintStream(new BufferedOutputStream(new FileOutputStream(FileDescriptor.out)), true));
+		}
 		/*[IF Sidecar19-SE_RAWPLUSJ9]*/
 		setIn(new BufferedInputStream(new FileInputStream(FileDescriptor.in)));
-		/*[ENDIF]*/
+		/*[ENDIF] Sidecar19-SE_RAWPLUSJ9 */
 		/*[ELSE]*/
 		/*[PR s66168] - ConsoleInputStream initialization may write to System.err */
 		/*[PR s73550, s74314] ConsolePrintStream incorrectly initialized */
+		/*[IF JAVA_SPEC_VERSION >= 11]*/
+		if (stderrCharset != null) {
+			hasSetErrEncoding = true;
+			setErr(com.ibm.jvm.io.ConsolePrintStream.localize(new BufferedOutputStream(new FileOutputStream(FileDescriptor.err)), true, stderrCharset));
+		} else {
+			setErr(com.ibm.jvm.io.ConsolePrintStream.localize(new BufferedOutputStream(new FileOutputStream(FileDescriptor.err)), true));
+		}
+		if (stdoutCharset != null) {
+			hasSetOutEncoding = true;
+			setOut(com.ibm.jvm.io.ConsolePrintStream.localize(new BufferedOutputStream(new FileOutputStream(FileDescriptor.out)), true, stdoutCharset));
```

Comment:
Now there are quite a few references of `com.ibm.jvm.io.ConsolePrintStream`, it can be imported.
Another suggestion is to make a helper method for `BufferedOutputStream` instantiation to reduce code duplication.

Reply:
I'll implement these in a separate PR.